app/controllers/certificates/pathways_controller.rb

```diff
@@ -7,10 +7,8 @@ def show
     @programme = @pathway.programme
     @current_user = current_user
 
-    recommended_activities = @pathway.pathway_activities.includes(:activity)
-    @recommended_community_activities = recommended_activities.filter { |pa| pa.activity.category == :community.to_s }
-    @recommended_community_activity_ids = @recommended_community_activities.map { _1.activity.id }
-    @recommended_activities = recommended_activities - @recommended_community_activities
+    @recommended_community_activity_ids = @pathway.pathway_activities.distinct.joins(:activity).where(activity: { category: :community }).pluck(:id)
+    @recommended_activities = @pathway.pathway_activities.includes(:activity).where.not(activity: { category: :community })
```

Comment:
I don't know if this is more efficient. But it is hugely more readable and maintainable. :+1: